sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/implementation/SyncTokenPolicyTest.java

```diff
@@ -0,0 +1,122 @@
+        // verify the new sync token value from the concurrent map
+        final String secondRequestHeader = response2.getRequest().getHeaders().getValue(SYNC_TOKEN);
+        assertEquals(constructSyncTokenStringWithoutSeqNumber(ID, VALUE), secondRequestHeader);
+
+        // verify the updated cached sync-token value
+        final HttpResponse response3 = pipeline2.send(new HttpRequest(HttpMethod.GET, new URL("http://localhost/"))).block();
+        final String thirdRequestHeader = response3.getRequest().getHeaders().getValue(SYNC_TOKEN);
+        assertEquals(constructSyncTokenStringWithoutSeqNumber(ID, UPDATED_VALUE), thirdRequestHeader);
+
+    }
+
+    @Test
+    public void multipleSyncTokenTest() throws Exception {
+        final SyncTokenPolicy syncTokenPolicy = new SyncTokenPolicy();
+
+        final String syncTokens = constructSyncTokenString(ID, VALUE, SEQUENCE_NUMBER) + "," + constructSyncTokenString(NEW_ID, UPDATED_VALUE, UPDATED_SEQUENCE_NUMBER);
+        final HttpPipeline pipeline = customizedPipeline(syncTokens, syncTokenPolicy);
+        final HttpResponse response = pipeline.send(new HttpRequest(HttpMethod.GET, new URL("http://localhost/"))).block();
+        // At first request, the request header should have empty sync-token value
+        final String firstRequestHeader = response.getRequest().getHeaders().getValue(SYNC_TOKEN);
+        assertEquals("", firstRequestHeader);
+        final String firstResponseHeader = response.getHeaders().getValue(SYNC_TOKEN);
+        assertEquals(syncTokens, firstResponseHeader);
+
+        final String secondSyncToken = constructSyncTokenString(ID, UPDATED_VALUE, UPDATED_SEQUENCE_NUMBER);
+        final HttpPipeline pipeline2 = customizedPipeline(secondSyncToken, syncTokenPolicy);
+        final HttpResponse response2 = pipeline2.send(new HttpRequest(HttpMethod.GET, new URL("http://localhost/"))).block();
+        // verify the new sync token value from the concurrent map
+        final String secondRequestHeader = response2.getRequest().getHeaders().getValue(SYNC_TOKEN);
+        final String secondRequestHeaderExpected = constructSyncTokenStringWithoutSeqNumber(ID, VALUE) + "," + constructSyncTokenStringWithoutSeqNumber(NEW_ID, UPDATED_VALUE);
+        assertEquals(secondRequestHeaderExpected, secondRequestHeader);
+
+        // verify the updated cached sync-token value
+        final HttpResponse response3 = pipeline2.send(new HttpRequest(HttpMethod.GET, new URL("http://localhost/"))).block();
+        final String thirdRequestHeader = response3.getRequest().getHeaders().getValue(SYNC_TOKEN);
+        final String thirdRequestHeaderExpected = constructSyncTokenStringWithoutSeqNumber(ID, UPDATED_VALUE) + "," + constructSyncTokenStringWithoutSeqNumber(NEW_ID, UPDATED_VALUE);
+        assertEquals(thirdRequestHeaderExpected, thirdRequestHeader);
+    }
+
+    private boolean syncTokenEquals(SyncToken syncToken, String id, String value, long sn) {
```

Comment:
Why does this return boolean? If the assertions fail, it'll never hit `return true;`